File: Problems 1900-1999/1910. Remove All Occurrences of a Substring/solution.py
class Solution:
    def removeOccurrences(self, s: str, part: str) -> str:
        stack = []
        partLen = len(part)
        for i, c in enumerate(s):
            stack.append(c)
            if i >= partLen - 1 and "".join(stack[-partLen:]) == part:
                for _ in range(partLen): stack.pop()

        return "".join(stack)

File: Problems 1900-1999/1910. Remove All Occurrences of a Substring/solution.cpp
class Solution {
public:
    string removeOccurrences(string s, string part) {
        int i = 0;
        int partLen = part.length();
        string res = s;
        for (char c : s) {
            res[i++] = c;
            if (i >= partLen && res.substr(i-partLen, partLen) == part) {
                i -= partLen;
            }
        }
        return res.substr(0, i);
    }
};